src/ui-kit/experimental/hierarchical/hierarchical/hierarchical.component.spec.ts

```diff
@@ -127,16 +124,58 @@ export class HierarchicalDataService implements SamHiercarchicalServiceInterface
   }
 
   getHiercarchicalById(item: SamHiercarchicalServiceSearchItem): Observable<SamHiercarchicalServiceResult> {
-    let data = Observable.of(this.loadedData);
+    let itemIncrease = 15;
+    let temp = this.getSortedData(this.loadedData, item.sort);
+    let data = Observable.of(temp);
+    let itemsOb: Observable<Object[]>;
     if (item.searchValue) {
-      return data.map(items =>
-        items.filter(itm => itm.parentId === item.id && (itm.name.indexOf(item.searchValue) !== -1
-          || itm.subtext.indexOf(item.searchValue) !== -1)));
+      itemsOb = data.map(items => items.filter(itm =>
+        itm.parentId === item.id &&
+        (itm.name.indexOf(item.searchValue) !== -1 ||
+          itm.subtext.indexOf(item.searchValue) !== -1
+        )));
     } else {
-      return data.map(items => items.filter(itm => itm.parentId === item.id));
+      itemsOb = data.map(items => items.filter(itm => itm.parentId === item.id));
+    }
+    let items: object[];
+    itemsOb.subscribe(
+      (result) => {
+        items = result;
+      }
+    );
+    let totalItemCount = items.length;
+
+    let maxSectionPosition = item.currentItemCount + itemIncrease;
+    if (maxSectionPosition > totalItemCount) {
+      maxSectionPosition = totalItemCount;
     }
+    let subItemsitems = items.slice(item.currentItemCount, maxSectionPosition);
+
+    let returnItem = {
+      items: subItemsitems,
+      totalItems: totalItemCount
+    };
+    return Observable.of(returnItem);
+  }
+
+
+  private getSortedData(data: any[], sort: Sort): any[] {
+    if (!sort || (!sort.active || sort.direction === '')) { return data; }
+    return data.sort((a, b) => {
+      let propertyA = this.sortingDataAccessor(a, sort.active);
+      let propertyB = this.sortingDataAccessor(b, sort.active)
+      const valueA = isNaN(+propertyA) ? propertyA : +propertyA;
+      const valueB = isNaN(+propertyB) ? propertyB : +propertyB;
+      return (valueA < valueB ? -1 : 1) * (sort.direction === 'asc' ? 1 : -1);
+    });
   }
 
+  private sortingDataAccessor: ((data: any, sortHeaderId: string) => string | number) =
```

Comment:
@thomastighe Isn't the initial type redundant? Couldn't you remote everything before the `=` and then remove the `=>` and just use it as a method?

Reply:
Cleaned this up